Spec/GCDTest.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  GCDTest.swift
+//  Ably
+//
+//  Created by Mikey on 12/01/2021.
+//  Copyright © 2021 Ably. All rights reserved.
+//
+import XCTest
+
+class GCDTest: XCTestCase {
+    func testScheduledBlockHandleDerefsBlockAfterInvoke() {
+        let invokedExpectation = self.expectation(description: "scheduled block invoked")
+        let obj = NSObject()
+
+        var scheduledBlock = artDispatchScheduled(0, .main) {
+            _ = obj
+            invokedExpectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 2, handler: nil)
+
+        _ = scheduledBlock // silence warning
+        scheduledBlock = nil
+
+        XCTAssertEqual(CFGetRetainCount(obj), 1)
```

Comment:
Check this in Instruments (like the original author did) is probably a good idea. Maybe request sample project from him?

Reply:
good idea, I emailed @mikepulaski for demo app.

Reply:
Sorry, I don't have an example project. However, if you run this test without the fix, it'll fail: any object(s) captured within the block will have a +1 retain count and will therefore never be deallocated.

Reply:
I understand. Thanks @mikepulaski you did a great job!